diff --git a/modules/system.views.inc b/modules/system.views.inc
index d49385c..9c29c6b 100644
--- a/modules/system.views.inc
+++ b/modules/system.views.inc
@@ -253,7 +253,6 @@ function system_views_data() {
       'base' => 'node',
       'base field' => 'nid',
       'relationship field' => 'id',
-      'extra' => array(array('field' => 'type', 'value' => 'node')),
     ),
   );
   $data['file_usage']['node_to_file'] = array(
